gulpfile.js

```diff
@@ -45,18 +48,57 @@ gulp.task('format', () => {
 });
 
 gulp.task('tsc', function(done) {
-  runSpawn(done, process.execPath, ['node_modules/typescript/bin/tsc']);
+  runSpawn(process.execPath, ['node_modules/typescript/bin/tsc'], done);
 });
 
 gulp.task('prepublish', function(done) {
   runSequence('tsc', 'copy', done);
 });
 
-gulp.task('default',['prepublish']);
-gulp.task('build',['prepublish']);
+gulp.task('default', ['prepublish']);
+gulp.task('build', ['prepublish']);
 
-gulp.task('test', ['format', 'build'], function(done) {
-  var opt_arg = [];
-  opt_arg.push('node_modules/jasmine/bin/jasmine.js');
-  runSpawn(done, process.execPath, opt_arg);
+// Command line commands
+gulp.task('update', ['build'], function(done) {
+  runSpawn(process.execPath, ['bin/webdriver-manager', 'update', '--android',
+      '--android-accept-licenses'], done);
 });
+gulp.task('start', ['build', 'shutdown'], function(done) {
+  runSpawn(process.execPath, ['bin/webdriver-manager', 'start', '--detach', '--seleniumPort',
+      '4444', '--android', '--appium-port', '4723', '--quiet'], done);
+});
+gulp.task('start:no_screen', ['build', 'shutdown'], function(done) {
+  runSpawn(process.execPath, ['bin/webdriver-manager', 'start', '--detach', '--seleniumPort',
+      '4444', '--android', '--appium-port', '4723', '--quiet', '--avds', 'none'], done);
+});
+gulp.task('shutdown', ['build'], function(done) {
+  runSpawn(process.execPath, ['bin/webdriver-manager', 'shutdown'], done);
+});
+
+// Test
+gulp.task('test:unit', ['build'], function(done) {
+  runSpawn(process.execPath, ['node_modules/jasmine/bin/jasmine.js'], done);
+});
+
+gulp.task('test:e2e:inner', ['build'], function(done) {
+  runSpawn(process.execPath, ['node_modules/jasmine/bin/jasmine.js', 'JASMINE_CONFIG_PATH=' +
+      path.join('e2e_spec', 'support', 'jasmine.json')], done);
+});
+gulp.task('test:e2e:inner:no_screen', ['build'], function(done) {
+  runSpawn(process.execPath, ['node_modules/jasmine/bin/jasmine.js', 'JASMINE_CONFIG_PATH=' +
+      path.join('e2e_spec', 'support', 'no_screen.json')], done);
+});
+gulp.task('test:e2e:no_update', function(done) {
+  runSequence('start', 'test:e2e:inner', 'shutdown', done);
+});
+gulp.task('test:e2e', function(done) {
+  runSequence('update', 'test:e2e:no_update', done);
+});
+gulp.task('test:e2e:no_screen', function(done) {
+  runSequence('update', 'start:no_screen', 'test:e2e:inner:no_screen', 'shutdown', done);
+});
+
+
+gulp.task('test', ['format', 'test:unit', 'test:e2e']);
+gulp.task('test:no_update', ['format', 'test:unit', 'test:e2e:no_update']);
+gulp.task('test:no_screen', ['format', 'test:unit', 'test:e2e:no_screen']);
```

Comment:
I think 'headless' is the more common term for this kind of thing than 'no_screen', but it's up to you.